src/cswinrt/strings/ComInteropHelpers.cs

```diff
@@ -485,3 +485,77 @@ public static DisplayInformation GetForMonitor(IntPtr monitor)
     }
 #endif
 }
+
+namespace Windows.ApplicationModel.DataTransfer
+{
+#if UAC_VERSION_1
+#if NET
+    [global::System.Runtime.Versioning.SupportedOSPlatform("Windows10.0.10240.0")]
+#endif
+#if EMBED
+    internal
+#else
+    public
+#endif
+    static class DataTransferManagerInterop
+    {
+        private static readonly global::System.Guid IDataTransferManagerInterop_IID = new global::System.Guid(0x3A3DCD6C, 0x3EAB, 0x43DC, 0xBC, 0xDE, 0x45, 0x67, 0x1C, 0xE8, 0x00, 0xC8);
+
+        private static readonly global::System.Guid riid = new global::System.Guid(0xA5CAEE9B, 0x8708, 0x49D1, 0x8D, 0x36, 0x67, 0xD2, 0x5A, 0x8D, 0xA0, 0x0C);
+
+#if NET
+        private static global::WinRT.IObjectReference objectReference => global::WinRT.ActivationFactory.Get<global::WinRT.IObjectReference>("Windows.ApplicationModel.DataTransfer.DataTransferManager", IDataTransferManagerInterop_IID);
+#else
+        private static global::WinRT.ObjectReference<global::WinRT.Interop.IUnknownVftbl> objectReference => global::WinRT.ActivationFactory.Get<global::WinRT.Interop.IUnknownVftbl>("Windows.ApplicationModel.DataTransfer.DataTransferManager", IDataTransferManagerInterop_IID);
+#endif
+
+        public static global::Windows.ApplicationModel.DataTransfer.DataTransferManager GetForWindow(global::System.IntPtr appWindow)
+        {
+            return IDataTransferManagerInteropMethods.GetForWindow(objectReference, appWindow, riid);
+        }
+
+        public static void ShowShareUIForWindow(global::System.IntPtr appWindow)
+        {
+            IDataTransferManagerInteropMethods.ShowShowShareUIForWindow(objectReference, appWindow);
+        }
+    }
+#endif
+
+    internal static class IDataTransferManagerInteropMethods
+    {
+#if NET
+        internal static unsafe global::Windows.ApplicationModel.DataTransfer.DataTransferManager GetForWindow(global::WinRT.IObjectReference _obj, global::System.IntPtr appWindow, in global::System.Guid riid)

```

Comment:
`ObjectReference` implements `IObjectReference` so we don't actually need the else case here or below.